prowler/lib/scan/scan.py

```diff
@@ -74,151 +72,190 @@ def __init__(
             ScanInvalidStatusError: If the status does not exist in the provider.
         """
         self._provider = provider
+        self._statuses = self._validate_statuses(status)
```

Comment:
I really like all this changes to encapsulate this logic but please move it to a different PR, we need also to cover the changes with tests now that we have several new methods for input validation.

This is one of the critical paths, we need to make sure that all changes are tested.